File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/Proxy/web/HideMeListProvider.java
package ru.kmorozov.gbd.core.logic.Proxy.web;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.nodes.TextNode;

import java.util.List;
import java.util.stream.Collectors;

/**
 * Created by km on 17.12.2016.
 */
public class HideMeListProvider extends AbstractProxyExtractor {

    private static final String PROXY_LIST_URL = "http://hideme.ru/proxy-list/?type=s#list";

    @Override
    protected String getProxyListUrl() {
        return HideMeListProvider.PROXY_LIST_URL;
    }

    @Override
    protected List<String> extractProxyList(Document doc) {
        return doc.getElementsByClass("proxy-table").get(0).
                getElementsByTag("tbody").get(0).getElementsByTag("tr").stream().map(HideMeListProvider::extractProxyData).collect(Collectors.toList());
    }

    private static String extractProxyData(Element element) {
        return ((TextNode) element.childNode(0).childNode(0)).getWholeText() + ':' + ((TextNode) element.childNode(1).childNode(0)).getWholeText();
    }
}

File: ru.kmorozov.gbd.core/src/main/java/ru/kmorozov/gbd/core/logic/proxy/web/HideMeListProvider.kt
package ru.kmorozov.gbd.core.logic.proxy.web

import org.jsoup.nodes.Document
import org.jsoup.nodes.Element
import org.jsoup.nodes.TextNode
import java.util.stream.Collectors

/**
 * Created by km on 17.12.2016.
 */
class HideMeListProvider : AbstractProxyExtractor() {

    override val proxyListUrl: String
        get() = PROXY_LIST_URL

    override fun extractProxyList(doc: Document): MutableList<String> {
        return doc.getElementsByClass("proxy-table")[0].getElementsByTag("tbody")[0].getElementsByTag("tr")
                .stream().map<String> { extractProxyData(it) }.collect(Collectors.toList())
    }

    companion object {

        private const val PROXY_LIST_URL = "http://hideme.ru/proxy-list/?type=s#list"

        private fun extractProxyData(element: Element): String {
            return (element.childNode(0).childNode(0) as TextNode).wholeText + ':'.toString() + (element.childNode(1).childNode(0) as TextNode).wholeText
        }
    }
}
